### PR TITLE
Feat/shipping refresh

### PR DESCRIPTION
**Pull request by joyet-simon**

### CODE REVIEW

File: cartridges/int_alma/cartridge/scripts/helpers/almaCheckoutHelper.js [hunk 1]
```diff
@@ -249,5 +249,6 @@ function formatPlanForCheckout(plan, currencyCode) {
 }
 
 module.exports = {
-    formatForCheckout: formatPlanForCheckout
+    formatForCheckout: formatPlanForCheckout,
```
Comment (Francois-Gomis) [inline cartridges/int_alma/cartridge/scripts/helpers/almaCheckoutHelper.js:252]: Why a name change ?

File: cartridges/int_alma/cartridge/templates/default/checkout/billing/paymentOptions/almaIncludes.isml [hunk 2]
```diff
@@ -1,41 +1,44 @@
...
-            dataUrl: "${pdict.data_url}",
-            createPaymentUrl: "${pdict.create_payment_url}",
-            checkoutFragmentUrl: "${pdict.checkout_fragment_url}"
-        }
-    }
+            dataUrl: '${pdict.data_url}',
+            createPaymentUrl: '${pdict.create_payment_url}',
+            checkoutFragmentUrl: '${pdict.checkout_fragment_url}',
+            orderAmountUrl: '${pdict.order_amount_url}',
+            getPlansRefresh: '${pdict.get_plans_refresh}'
```
Comment (Francois-Gomis) [inline cartridges/int_alma/cartridge/templates/default/checkout/billing/paymentOptions/almaIncludes.isml:29]: refresh

File: cartridges/int_alma/cartridge/scripts/helpers/almaPlanHelper.js [hunk 3]
```diff
@@ -113,45 +109,115 @@ function getPlansForWidget() {
...
 /**
  * Get data to initialize widget in cart and product detail
  * @param {string} locale e.g. "fr_FR"
  * @param {dw.order.Basket} currentBasket current basket
  * @returns {array} eligible data
  */
 function getPlansForCheckout(locale, currentBasket) {
-    var plansForEligibility = getAllowedPlans();
+    var feePlans = getFeePlans();
+
```
Comment (ClaireAlma) [inline cartridges/int_alma/cartridge/scripts/helpers/almaPlanHelper.js:145]: I think this function can be refactored, the code inside can be split in 3 functions at least

File: cartridges/int_alma/cartridge/static/default/js/almaCheckout.js [hunk 4]
```diff
@@ -1,13 +1,119 @@
...
+
+                    document.getElementById(`${plan.key + '_fragment'}`).innerHTML = "";
+
+                    if (plan.payment_plans) {
+                        document.getElementById(plan.key)
+                            .removeAttribute('hidden');
+                        document.getElementById(`${'alma-tab-' + plan.key + '-img'}`)
+                            .removeAttribute('hidden');
+
+                        var elementTabImg = document.getElementById(`${'alma-tab-' + plan.key + '-img'}`);
```
Comment (ClaireAlma) [inline cartridges/int_alma/cartridge/static/default/js/almaCheckout.js:41]: this function can be simplified

**Review by ClaireAlma — APPROVED** [review]

File: CHANGELOG.txt [hunk 5]
```diff
@@ -1,6 +1,13 @@
 CHANGELOG
 =========
 
+v4.1.0
+------
+* Fix: update price and informations on Alma payment method when shipping price change
+* Fix: display right Alma payment method for the current basket price
+* Fix: manage events on next step button in checkout
+* Fix: update custom data in Alma payload
```
Comment (Francois-Gomis) [inline CHANGELOG.txt:9]: do you have check custom data payload ? ( null )

File: cartridges/int_alma/cartridge/controllers/Checkout.js [hunk 6]
```diff
@@ -32,7 +32,8 @@ function getAlmaUrls() {
         data_url: URLUtils.http('Alma-BasketData').toString(),
         create_payment_url: URLUtils.https('Alma-CreatePaymentUrl').toString(),
         order_amount_url: URLUtils.http('Alma-OrderAmount').toString(),
-        checkout_fragment_url: URLUtils.http('Alma-FragmentCheckout').toString()
+        checkout_fragment_url: URLUtils.http('Alma-FragmentCheckout').toString(),
+        get_plans: URLUtils.http('Alma-GetPlans').toString()
```
Comment (Francois-Gomis) [inline cartridges/int_alma/cartridge/controllers/Checkout.js:36]: why get_plans ? plans ?

File: cartridges/int_alma/cartridge/scripts/helpers/almaPlanHelper.js [hunk 7]
```diff
@@ -113,45 +109,115 @@ function getPlansForWidget() {
...
+            }
+            if (plans[paymentMethod][keys].payment_plans) {
+                hasEligiblePaymentMethod = true;
+            }
+        });
+        formatedPaymentMethod.hasEligiblePaymentMethod = hasEligiblePaymentMethod;
+        formatedPaymentMethod.name = paymentMethod;
+        formatedPaymentMethod.plans = paymentMethodPlans;
+        formatedPlans.push(formatedPaymentMethod);
+    });
```
Comment (Francois-Gomis) [inline cartridges/int_alma/cartridge/scripts/helpers/almaPlanHelper.js:218]: to split

File: cartridges/int_alma/cartridge/controllers/Checkout.js (hunk 6 above)
Comment (joyet-simon) [reply cartridges/int_alma/cartridge/controllers/Checkout.js:36]: https://github.com/alma/sfcc-cartridge/pull/50#discussion_r1100131100

File: CHANGELOG.txt (hunk 5 above)
Comment (joyet-simon) [reply CHANGELOG.txt:9]: Yes and it's ok

File: cartridges/int_alma/cartridge/static/default/js/almaCheckout.js [hunk 8]
```diff
@@ -1,13 +1,97 @@
...
+            var data = await response.json()
+            almaPaymentMethods = data.plans;
+
+            for (const [indexPaymentMethod, almaPaymentMethod] of Object.entries(almaPaymentMethods)) {
+                var name = almaPaymentMethod.name;
+                var plans = almaPaymentMethod.plans;
+
+                for (const [indexPlan, plan] of Object.entries(plans)) {
+                    var icons = document.querySelectorAll(".alma-payment-method .fa");
+                    [].forEach.call(icons, function (icon) {
```
Comment (Benjamin-Freoua-Alma) [inline cartridges/int_alma/cartridge/static/default/js/almaCheckout.js:54]: It's forEach in empty array ?

**Review by Benjamin-Freoua-Alma — APPROVED** [review]

**Review by Francois-Gomis — APPROVED** [review]

**Pull request merged**